### PR TITLE
allow overriding a JS custom message handler; fixes #1419

### LINKED ISSUE #1419: Add 'override' argument to addCustomMessageHandler to allow overriding existing handler
Or perhaps name it `force` to be consistent with `shiny::registerInputHandler()`.

It's not a very important feature and likely won't be useful for too many people, but I've come across it on a few rare occasions. I can submit a PR if a shiny team member thinks this is acceptable.

Currently trying to redefine a JS message handler results in an error via https://github.com/rstudio/shiny/blob/9613c58bf8120bcfdab35801b17167418b5464ac/srcjs/shinyapp.js#L459-L461

### PR DESCRIPTION
**Pull request by daattali**
This commit assumes that "indexOf()` is available in javascript

### CODE REVIEW

**Comment by daattali** [comment]
Another note: this is my first time modifying javsacript code in shiny, so I hope I did the build process correctly

File: srcjs/shinyapp.js [hunk 1]
```diff
@@ -456,8 +456,13 @@ var ShinyApp = function() {
 
   // Adds custom message handler - this one is exposed to the user
   function addCustomMessageHandler(type, handler) {
+    // Remove any previously defined handlers so that only the most recent one
+    // will be called
     if (customMessageHandlers[type]) {
-      throw('handler for message of type "' + type + '" already added.');
+      var typeIdx = customMessageHandlerOrder.indexOf(type);
+      if (typeIdx != -1) {
```
Comment (wch) [inline srcjs/shinyapp.js]: Can you use `!==` here? I just realized that our eslint rules didn't check for this, and I've enabled that check on master.

**Comment by wch** [comment]
Please add a NEWS entry.

**Comment by daattali** [comment]
your comments were addressed

**Pull request merged**